s2s_client.go

```diff
@@ -0,0 +1,150 @@
+// Key is the Zignsec Authorization.
+// Timeout is only used by Authenticate() and Sign().
+func NewS2SClient(baseURL string, key string, timeout time.Duration) *S2SClient {
+	result := S2SClient{baseURL: baseURL, key: key, timeout: timeout}
+	return &result
+}
+
+// Authenticate request that does both Init() and Collect().
+func (c *S2SClient) Authenticate(nr string) (*CollectResponse, error) {
+	config := ZSInitConfig{Personalnumber: nr}
+	init, err := c.Init("Authenticate", config)
+	if err != nil {
+		return nil, err
+	}
+	return c.loop(init)
+}
+
+// Collect the state of an authenticate or sign request.
+// Order is from the Init() response.
+// This step is included in Authenticate() and Sign().
+// The guidelines recommend polling for results every 2 seconds.
+// When the response Status is COMPLETE, then further use of Collect() will receive an error.
+func (c *S2SClient) Collect(order string) (*CollectResponse, error) {
+	var result CollectResponse
+
+	url := c.baseURL + "/Collect?orderRef=" + order
+	req, err := http.NewRequest(http.MethodGet, url, nil)
+	if err != nil {
+		return nil, err
+	}
+	req.Header.Add("Authorization", c.key)
+	var httpClient http.Client
+	resp, err := httpClient.Do(req)
+	if err != nil {
+		return nil, err
+	}
+	b, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return nil, err
+	}
+	resp.Body.Close()
+	err = json.Unmarshal(b, &result)
+	return &result, err
+}
+
+// Init a authenticate or sign request.
+// Use Collect() to get the answer afterwards.
+// This step is included in Authenticate() and Sign().
+func (c *S2SClient) Init(method string, config ZSInitConfig) (*InitResponse, error) {
+	var result InitResponse
+
+	url := c.baseURL + "/" + method
+	configJSON, err := json.Marshal(config)
+	if err != nil {
+		return nil, err
+	}
+	req, err := http.NewRequest(http.MethodPost, url, bytes.NewReader(configJSON))
+	if err != nil {
+		return nil, err
+	}
+	setHeaders(req, c.key, "application/json")
+	var httpClient http.Client
+	resp, err := httpClient.Do(req)
+	if err != nil {
+		return nil, err
+	}
+	b, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return nil, err
+	}
+	resp.Body.Close()
+
+	err = json.Unmarshal(b, &result)
+	if err != nil {
+		return nil, err
+	}
+	return &result, nil
+}
+
+// Sign request that does both Init() and Collect().
+// It is sign that is signed. Prompt is for user guidance.
+// If prompt contains certain bytes (> 128?), Init() will fail with INVALID_PARAMETERS.
+func (c *S2SClient) Sign(Personalnumber string, prompt, sign []byte) (*CollectResponse, error) {
+	v := base64.StdEncoding.EncodeToString(prompt)
+	inv := base64.StdEncoding.EncodeToString(sign)
+	config := ZSInitConfig{Personalnumber: Personalnumber, UserVisibleData: v, UserNonVisibleData: inv}
+	init, err := c.Init("Sign", config)
+	if err != nil {
+		return nil, err
+	}
+	return c.loop(init)
+}
+
+//
+// Internal functions
+//
+
+func (c *S2SClient) loop(init *InitResponse) (*CollectResponse, error) {
```

Comment:
The library should not be responsible for polling every two seconds. The way this will work in a real scenario is

Client -> Auth-service -> Zignsec

So the client is responsible for polling, we just forward the request. So the loop logic should be removed altogether.